metadata.sh

```diff
@@ -135,6 +104,20 @@ for repo in "${repos[@]}"; do
 			continue
 		fi
 	fi
+
+	# check for categories that aren't in the canonical set
```

Comment:
This should be part of the `if` statement just above right? (so we're not checking `canonicalMetadataFile` against itself?)

Could probably then collapse the two `jq` invocations into one, but getting decent output out of that might be annoying?  Maybe have `jq` generate the error message string contents instead?